boxsdk/object/base_api_json_object.py

```diff
@@ -0,0 +1,53 @@
+# coding: utf-8
+
+from __future__ import unicode_literals, absolute_import
+
+import six
+
+from ..util.translator import Translator
+
+
+class BaseAPIJSONObjectMeta(type):
+    """
+    Metaclass for Box API objects. Registers classes so that API responses can be translated to the correct type.
+    Relies on the _item_type field defined on the classes to match the type property of the response json.
+    But the type-class mapping will only be registered if the module of the class is imported.
+    So it's also important to add the module name to __all__ in object/__init__.py.
+    """
+    def __init__(cls, name, bases, attrs):
+        super(BaseAPIJSONObjectMeta, cls).__init__(name, bases, attrs)
+        item_type = attrs.get('_item_type', None)
+        if item_type is not None:
+            Translator().register(item_type, cls)
+
+
+@six.add_metaclass(BaseAPIJSONObjectMeta)
+class BaseAPIJSONObject(object):
+    """Base class containing basic logic shared between true REST objects and other objects (such as an Event)"""
+
+    def __init__(self, response_object=None):
+        """
+        :param response_object:
+            The Box API response representing the object.
+        :type response_object:
+            :class:`BoxResponse`
```

Comment:
This is just a plain `dict`. `BoxResponse` is something else.

Looks like this was a typo in the original documentation. Can you fix it?

While we're at it, maybe we can give this a better description as well. Perhaps "A JSON object, from a Box API response, representing the object." or something like that.